.github/workflows/main-version-update.yml

```diff
@@ -0,0 +1,30 @@
+name: Update Main Version
```

Comment:
Heavily inspired by a github script used in `github-actions/checkout` - the repo is MIT licensed.